compiler/rustc_smir/src/rustc_smir/context.rs

```diff
@@ -587,6 +587,13 @@ impl<'tcx> Context for TablesWrapper<'tcx> {
         }
     }
 
+    fn intrinsic_name(&self, def: InstanceDef) -> Option<Symbol> {
+        let tables = self.0.borrow_mut();
+        let instance = tables.instances[def];
```

Comment:
This should be filtering intrinsics that have `InstanceDef::Intrinsic` or `InstanceDef::Item`. I don't believe it's correct to look at other instance kinds, though I'm not certain that there are any risks *currently*.

Reply:
Thanks! We're now filtering *after* the `tables` lookup, but let me know if you meant we should filter *before*. In that case, I think I'll have to pass the whole StableMIR instance.

Reply:
@compiler-errors I'm curious. Do we really need to include InstanceDef::Item here?

Reply:
The `Item` instance (of the same DefId as the `Intrinsic` instance) refers to the fallback body, yes. Codegen backends that do not support the intrinsic should always fall back to the default body and just call that. In that case it's not really an intrinsic anymore, so checking for intrinsics on the fallback bodies is likely not what is desirable most of the time.

Also you never get instances for fallback bodies unless you construct them yourself (the backends do that)

Reply:
Ok, well in this case that branch can be removed